src/app/modules/map/components/map-center-info/map-center-info.component.ts

```diff
@@ -187,29 +187,36 @@ export class MapCenterInfoComponent extends NgDestoryBase {
       'GET',
       `https://www.yr.no/api/v0/locations/search?language=nb&lat=${lat}&lon=${lon}&accuracy=100000`
     );
-    const apiResponse = await firstValueFrom(
-      this.http.request(apiReq).pipe(
-        filter((_r) => _r instanceof HttpResponse),
-        map((_r) => (_r as StrictHttpResponse<YrSearch>).body)
-      )
-    );
 
-    if (apiResponse.totalResults) {
-      const id = apiResponse._embedded.location[0].id;
-      const url =
-        {
-          nb: `https://www.yr.no/nb/v%C3%A6rvarsel/daglig-tabell/${id}`,
-          nn: `https://www.yr.no/nn/v%C3%AArvarsel/dagleg-tabell/${id}`,
-        }[this.translateService.currentLang] || `https://www.yr.no/en/forecast/daily-table/${id}`;
-      this.externalLinkService.openExternalLink(url);
-    } else {
-      const toastText = await firstValueFrom(this.translateService.get('MAP_CENTER_INFO.WEATHER_ERROR'));
-      const toast = await this.toastController.create({
-        message: toastText,
-        mode: 'md',
-        duration: 2000,
-      });
-      toast.present();
+    try {
+      const apiResponse = await firstValueFrom(
+        this.http.request(apiReq).pipe(
+          filter((_r) => _r instanceof HttpResponse),
+          map((_r) => (_r as StrictHttpResponse<YrSearch>).body)
+        )
+      );
+
+      if (apiResponse.totalResults) {
+        const id = apiResponse._embedded.location[0].id;
+        const url =
+          {
+            nb: `https://www.yr.no/nb/v%C3%A6rvarsel/daglig-tabell/${id}`,
+            nn: `https://www.yr.no/nn/v%C3%AArvarsel/dagleg-tabell/${id}`,
+          }[this.translateService.currentLang] || `https://www.yr.no/en/forecast/daily-table/${id}`;
+        this.externalLinkService.openExternalLink(url);
+      } else await this.toastOnYrError();
```

Comment:
Siden toastOnYrError uansett er feilhåndteringa i catch-blokken, så kan vi heller kaste en exception her? Det er også litt rar kode dette her, nesten overrasket over at ikke det feiler - er det lov å ha en statement rett etter else, altså ikke inne i en else-blokk?

Reply:
Det som jeg syns er rart er at `this.toastOnYrError()` kalles inne i en try-block der `this.toastOnYrError()` kalles i catch-blokken. Det betyr feks at `this.toastOnYrError()` kalles to ganger hvis det første kallet feiler...

Reply:
> Siden toastOnYrError uansett er feilhåndteringa i catch-blokken, så kan vi heller kaste en exception her? Det er også litt rar kode dette her, nesten overrasket over at ikke det feiler - er det lov å ha en statement rett etter else, altså ikke inne i en else-blokk?

Man trenger ikke å bruke block scope etter 'if else' hvis logikken tar bare en linje, men jeg kan selvfølgelig sette den for bedre forståelse.

Reply:
skal kaste error der ja, bra forslag